src/libcoretest/num/mod.rs

```diff
@@ -177,4 +177,61 @@ mod tests {
     test_impl_from! { test_u16i32, u16, i32 }
     test_impl_from! { test_u16i64, u16, i64 }
     test_impl_from! { test_u32i64, u32, i64 }
+
+    // Signed -> Float
+    test_impl_from! { test_i8f32, i8, f32 }
```

Comment:
I'd say conversions integer -> float feel... too suspicious for `Into`, even if they are lossless. I'd prefer to use more specialized methods/traits for this, at least for concrete types.
Do you have any example of generic code where this `Into` would be useful (I.e. a conversion from integer to float is required and it should be completely precise)?